tests/odbc/test_odbc.py

```diff
@@ -0,0 +1,104 @@
+import pandas as pd
+import psycopg2
+import pyodbc
+import pytest
+from pydantic import ValidationError
+
+from toucan_connectors.odbc.odbc_connector import OdbcConnector, OdbcDataSource
+
+
+def test_postgres_driver_installed():
+    """
+    This test aims to check that pgodbc is installed on the server
+    """
+    assert 'PostgreSQL Unicode' in pyodbc.drivers()
+
+
+@pytest.fixture(scope='module')
+def postgres_server(service_container):
+    def check(host_port):
+        conn = psycopg2.connect(
+            host='127.0.0.1',
+            port=host_port,
+            database='postgres_db',
+            user='ubuntu',
+            password='ilovetoucan',
+        )
+        cur = conn.cursor()
+        cur.execute('SELECT 1;')
+        cur.close()
+        conn.close()
+
+    return service_container('postgres', check, psycopg2.Error)
+
+
+@pytest.fixture
+def odbc_connector(postgres_server):
+    return OdbcConnector(
+        name='test',
+        connection_string=(
+            'DRIVER={PostgreSQL Unicode};'
+            'DATABASE=postgres_db;'
+            'UID=ubuntu;'
+            'PWD=ilovetoucan;'
+            'SERVER=127.0.0.1;'
+            'PORT=' + str(postgres_server['port']) + ';'
+        ),
+    )
+
+
+def test_invalid_connection_string():
+    """ It should raise an error as the connection string is invalid"""
+    with pytest.raises(ValidationError):
+        OdbcConnector(name='test')
+
+
+def test_raise_on_empty_query():
+    with pytest.raises(ValidationError):
+        OdbcDataSource(domain='test', name='test', database='postgres_db', query='')
+
+
+def test_odbc_get_df(mocker):
+    mockdbc = mocker.patch('pyodbc.connect')
+    mockdas = mocker.patch('pandas.read_sql')
```

Comment:
Don't hesitate to use variable identifiers that gives a good ideas of what's inside, and that are easily differentiable
```suggestion
    mock_pyodbc_connect = mocker.patch('pyodbc.connect')
    mock_pandas_read_sql = mocker.patch('pandas.read_sql')
```